src/templateBananaInBoxRule.ts

```diff
@@ -0,0 +1,65 @@
+import { BoundEventAst } from '@angular/compiler';
+import { IRuleMetadata, Replacement, RuleFailure } from 'tslint';
+import { AbstractRule } from 'tslint/lib/rules';
+import { SourceFile } from 'typescript';
+import { NgWalker } from './angular/ngWalker';
+import { BasicTemplateAstVisitor } from './angular/templates/basicTemplateAstVisitor';
+
+const INVALID_BOX = /\[(.*)\]/;
```

Comment:
`/\[(.*)\]/.test('[(ngModel)]') && /\[(.*)\]/.test('([ngModel])')` which seems incorrect.

Reply:
You're right, however, the previous regex (which one is different only by a question mark) was also wrong, it gave the same result for the tests you wrote above.

<hr>

After investigating why the specs pass even with this wrong regex, I discovered that the ast's name just cuts the parens, so:

**Test 1**: `<input type="text" name="foo" ([ngModel])="foo">`, test with regex returns true (the ast's name is just `[ngModel]`);
**Test 2**: `<div (emitter)="emitter"></div>` -> test with regex returns false (ast's name is `emitter`);
**Test 3**:  `<input type="text" name="foo" [ngModel]="foo">` -> test with regex returns false (ast's name is `ngModelChange`);
**Test 4**: `<input type="text" name="foo" [(ngModel)]="foo">` -> never call `visitEvent` method.

<hr>

I changed the regex a bit to: `/^\[(?!\()(.*)(?<!\))\]$/`.

Reply:
Yes, previous regex was absolutely wrong, I noticed that. Thanks for the update!

Reply:
yes @rafaelss95 , this rule used `visitEvent`, what explains this behavior.